src/main/java/appeng/me/cache/EnergyGridCache.java

```diff
@@ -508,8 +523,41 @@ public double getMaxStoredPower()
 	@Override
 	public double getEnergyDemand( final double maxRequired )
 	{
-		this.localSeen.clear();
-		return this.getEnergyDemand( maxRequired, this.localSeen );
+		final Queue<IEnergyGridProvider> toVisit = new LinkedList<>();
```

Comment:
Shouldn't a stack be enough here (for better performance)? Evaluation order would differ, but i don't think it matters.